### PR TITLE
Refactor symbolcontext

### PR DESCRIPTION
**Pull request by adrian-prantl**
depends on https://github.com/swiftlang/llvm-project/pull/9607

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by JDevlieghere — APPROVED** [review]
Beautiful.

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 1]
```diff
@@ -1752,17 +1752,16 @@ ConstString TypeSystemSwiftTypeRef::GetSwiftModuleFor(const SymbolContext &sc) {
...
-  return GetSwiftASTContextFromExecutionContext(nullptr);
+  if (!exe_scope)
+    return {};
+  ExecutionContext exe_ctx;
+  exe_scope->CalculateExecutionContext(exe_ctx);
+  return GetSymbolContext(&exe_ctx);
 }
 
-SwiftASTContext *TypeSystemSwiftTypeRef::GetSwiftASTContextFromExecutionContext(
+SymbolContext TypeSystemSwiftTypeRef::GetSymbolContext(
```
Comment (augusto2112) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:1764]: These two `TypeSystemSwiftTypeRef::GetSymbolContext` don't seem to have much to do with `TypeSystemSwiftTypeRef` (they also don't access any instance state and could be static), I wonder if these could live in somewhere more else so they could be more discoverable and reusable (for example, as static functions in `SymbolContext`)?

Comment (augusto2112) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:1764]: If we decide to do this these functions should probably de upstream.

**Pull request merged**